nupic/research/frameworks/dynamic_sparse/models/main.py

```diff
@@ -185,25 +191,42 @@ def has_params(self, module):
         elif isinstance(module, nn.Conv2d) and not self.sparse_linear_only:
             return "conv"
 
-    def save(self, checkpoint_dir):
-        # TODO: Implement
-        pass
+    def save(self, checkpoint_dir, experiment_name):
+        """
+        Save the model in this directory.
+        :param checkpoint_dir:
+        """
+        # experiment_root = os.path.join(checkpoint_dir, experiment_name)
+        # if not os.path.exists(experiment_root):
+        #     os.mkdir(experiment_root)
 
-    def restore(self, checkpoint_dir):
-        # TODO: Implement
-        pass
+        checkpoint_path = os.path.join(checkpoint_dir, experiment_name + ".pth")
+        torch.save(self.network.state_dict(), checkpoint_path)
 
-    # def _save(self, checkpoint_dir):
-    #     return self.model_save(checkpoint_dir)
+        config_file = os.path.join(checkpoint_dir, experiment_name + ".json")
+        with open(config_file, "w") as config_handler:
+            json.dump(self.config, config_handler)
 
-    # def _restore(self, checkpoint):
-    #     """Subclasses should override this to implement restore().
+        return
 
-    #     Args:
-    #         checkpoint (str | dict): Value as returned by `_save`.
-    #             If a string, then it is the checkpoint path.
-    #     """
-    #     self.model_restore(checkpoint)
+    def restore(self, checkpoint_dir, experiment_name):
+        """
+        :param checkpoint_path: Loads model from this checkpoint path.
+        If path is a directory, will append the parameter model_filename
+        """
+        # print("loading from", checkpoint_path)
+        # experiment_root = os.path.join(checkpoint_dir, experiment_name)
+        checkpoint_path = os.path.join(checkpoint_dir, experiment_name + ".pth")
+        self.network.load_state_dict(
+            torch.load(checkpoint_path, map_location=self.device)
+        )
+
+    def evaluate_noise(self, dataset):
```

Comment:
Not every model may require a noise dataset correct? It could be good to include logic that covers such cases.

Reply:
We discussed, but just to keep the record, the dataset is passed to the function so that should not be an issue.